### PR TITLE
Update dependencies to use ethers v6

### PR DESCRIPTION
**Pull request by fvictorio**
Hi, I updated the dependencies and code so that ethers v6 is used instead of v5.

As part of this I removed several explicit types, because I think the inferred types were correct. But after doing that I realized that maybe you prefer those explict types? If that's the case, let me know and I undo that part of the changes.

I also removed some dependencies that are no longer necessary (the ones under the `@ethersproject` org).

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -18,5 +18,5 @@
   },
   "exclude": ["node_modules"],
   "files": ["./hardhat.config.ts"],
-  "include": ["src/**/*", "tasks/**/*", "test/**/*", "deploy/**/*"]
+  "include": ["src/**/*", "tasks/**/*", "test/**/*", "deploy/**/*", "types/"]
```
Comment (fvictorio) [inline tsconfig.json:21]: This is necessary for things like `etyhers.getContractAt("Greeter", ...` to infer the proper types. It's also what typechain recommends (check the suggested tsconfig [here](https://github.com/dethcrypto/TypeChain/tree/master/packages/hardhat)).

**Comment by SvenMeyer** [comment]
@fvictorio 
1) It would also be nice to support the build-in `bigint` type which ethers v6 and typescript support. It just complains that the target has to be > es6, so some newer version would need to be set for `target` (and `lib` ?) in file `tsconfig.json`

2) Not sure if it was a new issue, but `pnpm coverage` caused an error 
`Error HH700: Artifact for contract "hardhat/console.sol:console" not found. `
Doing `npx hardhat clean` fixed it for me, so I added that into `package.json` as well.
`"coverage": "hardhat clean && hardhat coverage --solcoverjs ./.solcover.js --temp artifacts --testfiles \"test/**/*.ts\" && pnpm typechain",`

3) Beside that, I made all the same changes in an existing fresh (from a few days ago) project and it works for me.

**Comment by fvictorio** [comment]
> * It would also be nice to support the build-in `bigint` type which ethers v6 and typescript support. It just complains that the target has to be > es6, so some newer version would need to be set for `target` (and `lib` ?) in file `tsconfig.json`

Great catch, I updated the `tsconfig.json` file. Now bigint literals should work.

> Not sure if it was a new issue, but pnpm coverage caused an error

This doesn't happen to me :/

> 3\. Beside that, I made all the same changes in an existing fresh (from a few days ago) project and it works for me.

Nice, glad to hear that!

**Comment by PaulRBerg** [comment]
Thanks a lot for this PR, @fvictorio, and for your feedback, @SvenMeyer.

CI is passing, so I'm happy to merge this right away.

**Comment by fvictorio** [comment]
Thanks!

**Pull request merged**